httpx/router.go

```diff
@@ -38,7 +41,7 @@ func (r *Router) HandleFunc(path string, f func(context.Context, http.ResponseWr
 
 // Header adds a route that will be used if the header value matches.
 func (r *Router) Headers(pairs ...string) *Route {
-	return &Route{r.mux.Headers(pairs...)}
+	return r.getOrCreateRoute(r.mux.Headers(pairs...), "")
```

Comment:
Oh, missed your comment about not modifying the route method. That does solve the edge case I mentioned, but my hope was that this would eventually expose everything that gorilla mux exposes.